service/frontend/workflowHandler.go

```diff
@@ -155,13 +158,22 @@ func NewWorkflowHandler(sVice service.Service, config *Config, metadataMgr persi
 
 // Start starts the handler
 func (wh *WorkflowHandler) Start() error {
-	wh.Service.GetDispatcher().Register(workflowserviceserver.New(wh))
+	currentClusteName := wh.GetClusterMetadata().GetCurrentClusterName()
+	dcRedirectionPolicy := RedirectionPolicyGenerator(
+		wh.GetClusterMetadata(),
+		wh.domainCache,
+		wh.dcRedirectionPolicy,
+	)
+	dcRediectionHandle := NewDCRedirectionHandler(
+		currentClusteName, dcRedirectionPolicy, wh.Service, wh,
+	)
+	wh.Service.GetDispatcher().Register(workflowserviceserver.New(dcRediectionHandle))
```

Comment:
I'm a little worried that we always register the redirect handler.  I think RedirectHandler needs to wrap WorkflowHandler and only registered when we need forwarding.

Reply:
take a look at the noop redirection policy, which is only 6 lines of code (see below).
if we trust that the redirection handler works, we should also trust the code below.

```
// GetTargetDatacenterByName get target cluster name by domain Name
func (policy *NoopRedirectionPolicy) GetTargetDatacenterByName(domainName string) (string, error) {
	return policy.currentClusteName, nil
}

// GetTargetDatacenterByID get target cluster name by domain ID
func (policy *NoopRedirectionPolicy) GetTargetDatacenterByID(domainID string) (string, error) {
	return policy.currentClusteName, nil
}
```